### ISSUE TITLE
build: SystemJS and script tags

### ISSUE THREAD

**Issue by wojciechpolak** [FountainJS/generator-fountain-webapp#22]
Hi,

While doing a build, systemjs version replaces all `script` tags in `index.html` (gulp_tasks/systemjs.js):

```
function updateIndexHtml() {
  return gulp.src(conf.path.src('index.html'))
    .pipe(replace(
      /<script[\s\S]*script>/,
      `<script src="index.js"></script>`
    ))
    .pipe(gulp.dest(conf.path.tmp()));
}
```

The problem is that this also removes all custom `script` tags, for example I want to use a script with a type `text/ng-template`. While having this, that simple regex breaks the final build -- the `index.js` in `dist` is never created.

As a workaround to this I wrapped the html code and changed the updateIndexHtml().

index.html:

```
    <!-- systemjs:code -->
    <!-- build:js(.tmp) index.js -->
    <script src="jspm_packages/system.js"></script>
    <script src="jspm.browser.js"></script>
    <script src="jspm.config.js"></script>

    <script>
      System.import('src/index.js');
    </script>
    <!-- endbuild -->
    <!-- systemjs:endcode -->
```

systemjs.js:

```
function updateIndexHtml() {
  return gulp.src(conf.path.src('index.html'))
    .pipe(replace(
      /<!-- systemjs:code -->[\s\S]*<!-- systemjs:endcode -->/,
      `<!-- build:js(.tmp) index.js -->
    <script src="index.js"></script>
    <!-- endbuild -->`
    ))
    .pipe(gulp.dest(conf.path.tmp()));
}
```

Please consider this or some better solution.

**Comment by Swiip** [comment]
Thanks for reporting. I think you're right, our regex is way too generic. I would hope to find a solution without involving comments but it has to be better than it is.

**Comment by wojciechpolak** [comment]
Uhm, @Swiip did you test your change in practice?
Now I get this error:

```
[11:06:10] Error: Error: File not found with singular glob: [path]/.tmp/jspm_packages/system.js
    at DestroyableTransform.<anonymous> ([path]/node_modules/gulp-useref/lib/streamManager.js:90:36)
    at emitOne (events.js:95:20)
    at DestroyableTransform.emit (events.js:182:7)
    at emitOne (events.js:95:20)
    at Through2.emit (events.js:182:7)
    at OrderedStreams.<anonymous> ([path]/node_modules/glob-stream/index.js:138:20)
    at emitOne (events.js:90:13)
    at OrderedStreams.emit (events.js:182:7)
    at emitOne (events.js:90:13)
    at DestroyableTransform.emit (events.js:182:7)
    at Glob.<anonymous> ([path]/node_modules/glob-stream/index.js:38:16)
    at Glob.g (events.js:273:16)
    at emitOne (events.js:90:13)
    at Glob.emit (events.js:182:7)
    at Glob._finish ([path]/node_modules/glob-stream/node_modules/glob/glob.js:172:8)
    at done ([path]/node_modules/glob-stream/node_modules/glob/glob.js:159:12)
    at Glob._processSimple2 ([path]/node_modules/glob-stream/node_modules/glob/glob.js:652:12)
    at [path]/node_modules/glob-stream/node_modules/glob/glob.js:640:10
    at Glob._stat2 ([path]/node_modules/glob-stream/node_modules/glob/glob.js:736:12)
    at lstatcb_ ([path]/node_modules/glob-stream/node_modules/glob/glob.js:728:12)
    at RES ([path]/node_modules/inflight/inflight.js:23:14)
    at f ([path]/node_modules/once/once.js:17:25)

```

**Comment by micaelmbagira** [comment]
Which framework/modules/js are you using ? Are you using `master` branch ?

**Comment by wojciechpolak** [comment]
I'm using v0.4.0 (Angular1, ES6, SystemJS) with your latest change to this issue.

**Comment by micaelmbagira** [comment]
On v0.4.0, `gulp serve:dist` is not working. Next release will fix it, meanwhile you can check out the changes on master.

**Comment by Swiip** [comment]
I must admit I didn't test much no. But I don't see the link of my change and your log.

**Comment by wojciechpolak** [comment]
@micaelmbagira I am aware of the issues with serve:dist and I already mixed v0.4.0 with some later changes to the master. The stack trace I reported above has started after the last @Swiip change to this issue. I don't see the strong link either, but when I revert the change, there's no problem. So there is some mystery link ;)

One more thing I noticed after the last change. This is the result now in `.tmp/index.html`:

```
    <!-- build:js(.tmp) index.js -->
    <script src="jspm_packages/system.js"></script>
    <script src="jspm.browser.js"></script>
    <script src="jspm.config.js"></script>

    <script src="index.js"></script>
    <!-- endbuild -->
```

Previously only the index.js remained and now jspm things stay also.
